pretext/build.py

```diff
@@ -40,6 +40,9 @@ def html(
                 None,
                 output.as_posix(),
             )
+            codechat.map_path_to_xml_id(
```

Comment:
Do we want creating the mapping.json to be optional? How expensive is the process?

Reply:
The amount of work seems fairy minimal compared to the processing of a typical build to me: it just re-parses the XML, then looks only at XML xml:id attributes. If I can, I'll try collecting some profile data. What do you think is a good test case -- which book should I build to evaluate performance?